### PR TITLE
Trac 34738: Customizer preview elements lose or misplace their preview data

### PR DESCRIPTION
**Pull request by westonruter**
https://core.trac.wordpress.org/ticket/34738

- [x] Add unit tests.

### CODE REVIEW

File: src/wp-includes/class-wp-customize-setting.php [hunk 1]
```diff
@@ -183,6 +192,9 @@ protected function aggregate_multidimensional() {
 			self::$aggregated_multidimensionals[ $this->type ] = array();
 		}
 		if ( ! isset( self::$aggregated_multidimensionals[ $this->type ][ $id_base ] ) ) {
+			// Only add one action per multidimensional value.
```
Comment (westonruter) [inline src/wp-includes/class-wp-customize-setting.php]: Only add one action per multidimensional ID base.

File: src/wp-includes/class-wp-customize-setting.php [hunk 2]
```diff
@@ -183,6 +192,9 @@ protected function aggregate_multidimensional() {
...
+			add_action( 'customize_post_value_set', array( $this, '_clear_aggregated_multidimensional_perview_applied_flag' ) );
```
Comment (westonruter) [inline src/wp-includes/class-wp-customize-setting.php]: Spelling mistake.

Comment (westonruter) [inline src/wp-includes/class-wp-customize-setting.php]: Should we just use `customize_post_value_set_{$this->id}` instead? That will result in fewer actions triggered, but more actions added.

**Comment by marcperel** [comment]
I've run this a few times running Twenty16 and LayersWP and it checks out for me. Good work Weston.

**Comment by westonruter** [comment]
@marcperel thank you very much for testing. This has been committed to `trunk`: https://core.trac.wordpress.org/changeset/35724

**Pull request closed without merging**